modules/tinymce/src/plugins/table/test/ts/browser/ui/TableDialogStyleWithCssTest.ts

```diff
@@ -0,0 +1,200 @@
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/table/Plugin';
+import Theme from 'tinymce/themes/silver/Theme';
+
+import * as TableTestUtils from '../../module/test/TableTestUtils';
+
+describe('browser.tinymce.plugins.table.ui.TableCellDialogTest', () => {
+  const generalSelectors = {
+    cellspacing: 'label.tox-label:contains(Cell spacing) + input.tox-textfield',
+    cellpadding: 'label.tox-label:contains(Cell padding) + input.tox-textfield'
+  };
+
+  const initializeTable = (editor: Editor, tableAttributes: Record<string, string>, tableStyles: Record<string, string>, cellStyles: Record<string, string>) => {
+    const defaultTable = SugarElement.fromHtml<HTMLTableElement>(
+      '<table>' +
+      '<tbody>' +
+      '<tr>' +
+      '<td><p>Hello</p></td>' +
+      '<td><p>Next cell</p></td>' +
+      '</tr>' +
+      '<tr>' +
+      '<td><p>Cell 3</p></td>' +
+      '<td><p>Last one</p></td>' +
+      '</tr>' +
+      '</tbody>' +
+      '</table>'
+    );
+
+    Attribute.setAll(defaultTable, tableAttributes);
+    Css.setAll(defaultTable, tableStyles);
+
+    Arr.each(SelectorFilter.descendants(defaultTable, 'td,th'), (elem) => Css.setAll(elem, cellStyles));
+
+    editor.setContent(Html.getOuter(defaultTable));
+    TinySelections.setCursor(editor, [ 0, 0, 0 ], 0);
+  };
+
+  const assertTable = (editor: Editor, tableAttributes: Record<string, string>, tableStyles: Record<string, string>, cellStyles: Record<string, string>) => {
+    TinyAssertions.assertContentStructure(editor, ApproxStructure.build((s, str, _arr) => {
+      const cell = s.element('td', {
+        styles: Obj.map(cellStyles, (val) => val === '' ? str.none() : str.is(val))
+      });
+
+      const row = s.element('tr', { children: [ cell, cell ] });
+      const tbody = s.element('tbody', { children: [ row, row ] });
+
+      const table = s.element('table', {
+        children: [ tbody ],
+        attrs: Obj.map(tableAttributes, (val) => val === '' ? str.none() : str.is(val)),
+        styles: Obj.map(tableStyles, (val) => val === '' ? str.none() : str.is(val))
+      });
+
+      return s.element('body', { children: [ table ] });
+    }));
+  };
+
+  Arr.each([
+    { title: 'using attributes', style_by_css: false },
+    { title: 'using styles', style_by_css: true }
+  ], (spec) => {
+    context(`Control table layout ${spec.title}`, () => {
+      const hook = TinyHooks.bddSetup<Editor>({
+        plugins: 'table',
+        base_url: '/project/tinymce/js/tinymce',
+        toolbar: 'tableprops',
+        table_style_by_css: spec.style_by_css
+      }, [ Plugin, Theme ]);
+
+      it('falls back to cellpadding attribute if no CSS is defined', async () => {
+        const editor = hook.editor();
+        initializeTable(editor, { cellpadding: '20px' }, {}, {});
+
+        await TableTestUtils.pOpenTableDialog(editor);
+
+        TableTestUtils.assertDialogValues({
```

Comment:
I'd wrap `asserDialogValues` in a function and have inside the `false` and `generalSelectors` so you don't need to pass them every time as only the first param changes?!